docs/source/cpp/compute.rst

```diff
@@ -1474,13 +1474,15 @@ value, but smaller than nulls.
 Structural transforms
 ~~~~~~~~~~~~~~~~~~~~~
 
-+--------------------------+------------+--------------------+---------------------+---------+
-| Function name            | Arity      | Input types        | Output type         | Notes   |
-+==========================+============+====================+=====================+=========+
-| list_flatten             | Unary      | List-like          | List value type     | \(1)    |
-+--------------------------+------------+--------------------+---------------------+---------+
-| list_parent_indices      | Unary      | List-like          | Int32 or Int64      | \(2)    |
-+--------------------------+------------+--------------------+---------------------+---------+
++--------------------------+------------+--------------------------+---------------------+---------+
+| Function name            | Arity      | Input types              | Output type         | Notes   |
++==========================+============+==========================+=====================+=========+
+| list_flatten             | Unary      | List-like                | List value type     | \(1)    |
++--------------------------+------------+--------------------------+---------------------+---------+
+| list_parent_indices      | Unary      | List-like                | Int32 or Int64      | \(2)    |
++--------------------------+------------+--------------------------+---------------------+---------+
+| list_element             | Binary     | List-like, Int32 or Int64| List value type     | \(3)    |
```

Comment:
For _Input types_ instead of using a comma, you can use this form: `List-like (Arg 0); Integral (Arg 1)`.
Refer to _case_when_ doc.

Reply:
Also, you need to add the `list_element` kernel to [Python's API doc](https://github.com/apache/arrow/blob/master/docs/source/python/api/compute.rst).

Reply:
Also try to keep things in alphabetical order, and reformat the table as needed (Input types could have one more column of padding added to it).